### PR TITLE
Add trajectory_parameterization parameter to CartesianPath & JointInterpolationPlanner

### PR DESCRIPTION
**Pull request by JafarAbdi**
Fix: https://github.com/ros-planning/moveit_task_constructor/issues/330

Depend on https://github.com/ros-planning/moveit/pull/3021

I think this's simpler than the plugin based solution

### CODE REVIEW

**Review by rhaschke — COMMENTED** [review]
> I think this's simpler than the plugin-based solution.

Sure, it's easier, but less flexible! A plugin-based solution would allow using the other time parameterization methods for MoveIt planning pipelines as well.

File: core/include/moveit/task_constructor/solvers/cartesian_path.h [hunk 1]
```diff
@@ -50,7 +51,8 @@ MOVEIT_CLASS_FORWARD(CartesianPath);
 class CartesianPath : public PlannerInterface
 {
 public:
-	CartesianPath();
+	explicit CartesianPath(std::unique_ptr<trajectory_processing::TimeParameterization> trajectory_parameterization =
+	                           std::make_unique<trajectory_processing::IterativeParabolicTimeParameterization>());
```
Comment (AndyZe) [inline core/include/moveit/task_constructor/solvers/cartesian_path.h:55]: No comment on the high-level strategy. But please make the default TOTG

**Comment by JafarAbdi** [comment]
> A plugin-based solution would allow using the other time parameterization methods for MoveIt planning pipelines as well.

I thought this's already the case [TimeOptimalTrajectoryGeneration](https://github.com/ros-planning/moveit/blob/master/moveit_ros/planning/planning_request_adapter_plugins/src/add_time_optimal_parameterization.cpp), [IterativeParabolicTimeParameterization](https://github.com/ros-planning/moveit/blob/master/moveit_ros/planning/planning_request_adapter_plugins/src/add_time_parameterization.cpp), [IterativeSplineParameterization](https://github.com/ros-planning/moveit/blob/master/moveit_ros/planning/planning_request_adapter_plugins/src/add_iterative_spline_parameterization.cpp) am I missing something ?

**Review by rhaschke — APPROVED** [review]
> I thought this's already the case TimeOptimalTrajectoryGeneration, IterativeParabolicTimeParameterization, IterativeSplineParameterization.

You are right. I was misled by the too generic name `AddTimeParameterization` used for the IPTP plugin...

Approving. Depends on https://github.com/ros-planning/moveit/pull/3021 and https://github.com/ros-planning/moveit/pull/3023.

**Comment by AndyZe** [comment]
I've tested this. It works well.  :+1:

**Review by AndyZe — APPROVED** [review]

**Review by rhaschke — COMMENTED** [review]
@JafarAbdi, may I ask you to change the API such that the time parameterization is specified as a property of those classes?
Time parameterization is also used by `merge(...)`:
https://github.com/ros-planning/moveit_task_constructor/blob/4fa8c10f4456e893e09212b51a06b3d2ec4e9266/core/src/merge.cpp#L168-L170
which is used by `Connect` and `Merger` stages. Do you mind adding this feature there as well?

**Review by henningkayser — APPROVED** [review]
This looks very straight-forward to me. I agree it makes much more sense than using plugins here. Do you think you could address Robert's comments so that this can get merged?

**Pull request closed without merging**